pkg/api/imagebasedgroupupgrades/v1alpha1/types.go

```diff
@@ -0,0 +1,104 @@
+package v1alpha1
+
+import (
+	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:generate=true
+
+// ImageBasedUpgradeAction defines the type of the action to perform. Abort will stop the upgrade by setting the IBU stage to Idle. Finalize will remove the previous stateroot after upgrade is done by setting the IBU stage to Idle. Prep will start preparing the upgrade by setting the IBU stage to Prep. Upgrade will start the upgrade process by setting the IBU stage to Upgrade. Rollback will pivot back to previous stateroot by setting the IBU stage to Rollback.
+type ImageBasedUpgradeAction struct {
+	// +kubebuilder:validation:Enum=Abort;Prep;Upgrade;Rollback;Finalize
+	Name string `json:"name"`
```

Comment:
name sounds like a freeform string. maybe a different field name that implies enum?